nixos/modules/programs/environment.nix

```diff
@@ -18,7 +18,8 @@ in
 
     environment.variables =
       { NIXPKGS_CONFIG = "/etc/nix/nixpkgs-config.nix";
-        PAGER = mkDefault "less -R";
+        PAGER = mkDefault "less";
+        LESS = mkDefault "-R";
```

Comment:
What is the effect of the new `LESS` environment variable on configurations that override `PAGER` to `less` with no options or other options?

Reply:
They get -R added aka ``--RAW-CONTROL-CHARS``.

Reply:
That's technically a breaking change then, although unlikely.

Reply:
Especially with git. From the previous PR:

https://github.com/NixOS/nixpkgs/pull/13470#issuecomment-189076451

Reply:
I definitely found this to cause a disconcerting change in `git`'s behaviour.  Without `-F` and `-X`, git commands produce much more of an interruption to the flow of my terminal.  There's no way to get the output of multiple git commands on the screen at once, or look at the output of a git command right above/below the output of some other shell command, etc. etc.

[This excellent SO post](https://unix.stackexchange.com/questions/469360/less-performs-differently-when-invoked-from-bash-and-from-git) points to the documentation for the `core.pager` config key, which says:

> When the LESS environment variable is unset, Git sets it to FRX (if LESS environment variable is set, Git does not change it at all).

What we need here is a way to specify a lower-priority default value for the `LESS` environment variable.  Fortunately, `lesskey` seems to be exactly that.  I think if we used the [`programs.less.envVariables`](https://github.com/NixOS/nixpkgs/blob/master/nixos/modules/programs/less.nix#L82) option to set `LESS=-R`, then programs like `dmesg` that just launch `less` with no args will get the `-R` option, while `git` will still set `LESS` to its preferred value.

I haven't tried this out yet but hope to in the next couple of days.

Reply:
https://github.com/NixOS/nixpkgs/pull/139988 is what I arrived at, FYI.

Reply:
Same, the lack of `-F` means that an empty `git diff` will show a blank page.

This PR was merged too fast.

Reply:
@zimbatm I think you're right. What do you think of #139988?